src/components/tabs/__internal__/tabs-header/tabs-header.component.tsx

```diff
@@ -40,61 +33,115 @@ const TabsHeader = ({
   extendedLine,
   noRightBorder = false,
   isInSidebar = false,
+  size = "default",
 }: TabHeaderProps) => {
-  const [leftScrollOpacity, setLeftScrollOpacity] = useState(0);
-  const [rightScrollOpacity, setRightScrollOpacity] = useState(1);
-
   const ref = useRef<HTMLDivElement>(null);
-
-  let isScrollable = false;
-
   const { current } = ref;
+  const [leftVisible, setLeftVisible] = useState(false);
+  const [rightVisible, setRightVisible] = useState(false);
 
-  if (position === "top" && current) {
-    isScrollable = current.scrollWidth > current.clientWidth;
-  }
+  const updateUI = useCallback(() => {
```

Comment:
**issue**: I noticed some visual glitches when a tab title, partially covered by a navigation button, seems to overlap the button when focused or if it has a validation border.

Here's an example when a covered tab title is focused: 

https://github.com/user-attachments/assets/4f2276b6-ab7e-4e42-a344-c2cf345f9891

And another when a covered tab title has a validation border:

![navigation-button-validation-border](https://github.com/user-attachments/assets/57ca7ba7-b7c1-4a5c-938f-7ae6ca8d9295).

To avoid this glitch, perhaps we could adjust the scrolling behaviour to ensure that a partially covered tab title is fully in view when focused. What do you think?